src/features/copy-paste/copy-paste-context-menu.ts

```diff
@@ -0,0 +1,104 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { inject, injectable } from "inversify";
+import {
+    Action,
+    IActionDispatcher,
+    IActionHandler,
+    IContextMenuItemProvider,
+    isDeletable,
+    isSelected,
+    MenuItem,
+    Point,
+    SModelRoot,
+    TYPES
+} from "sprotty/lib";
+
+import { GLSPServerStatusAction } from "../../model-source/glsp-server-status";
+import { GLSP_TYPES } from "../../types";
+import { SelectionService } from "../select/selection-service";
+
+export class InvokeCopyAction implements Action {
+    static readonly KIND = "invoke-copy";
+    kind = InvokeCopyAction.KIND;
+}
+
+export class InvokePasteAction implements Action {
+    static readonly KIND = "invoke-paste";
+    kind = InvokePasteAction.KIND;
+}
+
+export class InvokeCutAction implements Action {
+    static readonly KIND = "invoke-cut";
+    kind = InvokeCutAction.KIND;
+}
+
+@injectable()
+export class InvokeCopyPasteActionHandler implements IActionHandler {
+    @inject(TYPES.IActionDispatcher) protected dispatcher: IActionDispatcher;
+    handle(action: Action) {
+        switch (action.kind) {
+            case InvokeCopyAction.KIND:
+                document.execCommand('copy');
+                break;
+            case InvokePasteAction.KIND:
+                // in a browser without additional permission we can't invoke the paste command
+                // the user needs to invoke it from the browser, so notify the user about it
+                this.dispatcher.dispatch(
+                    <GLSPServerStatusAction>{
```

Comment:
This shows a warning directly in the GLSP editor which is VERY subtle, I almost did not notice it. In Theia they open a message box. Not sure which way we should go with that but I just wanted to point that out ;-)

Reply:
Yes, thanks for raising that! This should actually be addressed with https://github.com/eclipse-glsp/glsp-theia-integration/pull/24